File: kotlin/functions/list_operations.kt
fun main() {
    val numbers = mutableListOf(1, 2, 3, 4, 5)

    println("Numbers: $numbers")
    println("Size: ${numbers.size}")
    println("Sum: ${numbers.sum()}")
    println("Max: ${numbers.maxOrNull()}")
    println("Min: ${numbers.minOrNull()}")

    numbers.add(6)
    println("After add: $numbers")

    numbers.removeLast()
    println("After removeLast: $numbers")

    val squared = numbers.map { it * it }
    println("Squared: $squared")

    val evens = numbers.filter { it % 2 == 0 }
    println("Even numbers: $evens")

    val sorted = numbers.sortedDescending()
    println("Sorted descending: $sorted")

    println("First element: ${numbers.first()}")
    println("Last element: ${numbers.last()}")

    val unique = listOf(1, 2, 2, 3, 3, 3).distinct()
    println("Unique: $unique")
}

File: csharp/functions/ListOperations.cs
using System;
using System.Collections.Generic;
using System.Linq;

class ListOperations
{
    static void Main()
    {
        List<int> numbers = new List<int> { 1, 2, 3, 4, 5 };

        Console.WriteLine($"Numbers: {string.Join(", ", numbers)}");
        Console.WriteLine($"Count: {numbers.Count}");
        Console.WriteLine($"Sum: {numbers.Sum()}");
        Console.WriteLine($"Max: {numbers.Max()}");
        Console.WriteLine($"Min: {numbers.Min()}");

        numbers.Add(6);
        Console.WriteLine($"After Add: {string.Join(", ", numbers)}");

        numbers.RemoveAt(numbers.Count - 1);
        Console.WriteLine($"After Remove: {string.Join(", ", numbers)}");

        var squared = numbers.Select(x => x * x).ToList();
        Console.WriteLine($"Squared: {string.Join(", ", squared)}");

        var evens = numbers.Where(x => x % 2 == 0).ToList();
        Console.WriteLine($"Even numbers: {string.Join(", ", evens)}");
    }
}
